magefile.go

```diff
@@ -104,3 +124,23 @@ func Precommit() error {
 func Check() {
 	mg.SerialDeps(Lint, Test)
 }
+
+// BuildExamples builds all the examples
+func BuildExamples() error {
+	return filepath.WalkDir("./examples/", func(path string, d os.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() && d.Name() != "examples" {
```

Comment:
Instead of `d.Name() != "examples"`, we could try checking for the existence of `filepath.Join(path, "go.mod")`